tests/unit/history/createHashHistory.ts

```diff
@@ -47,12 +47,17 @@ suite('createHashHistory', () => {
 
 	test('emits change when path is updated', () => {
 		const history = createHashHistory({ window: sandbox.contentWindow });
-		let emittedValue = '';
+		let emittedValues: string[] = [];
 		history.on('change', ({ value }) => {
-			emittedValue = value;
+			emittedValues.push(value);
 		});
 		history.set('/foo');
-		assert.equal(emittedValue, '/foo');
+
+		return new Promise((resolve) => setTimeout(resolve, 500))
```

Comment:
The underlying `hashchange` event is emitted asynchronously. That's why the test didn't originally catch the bug.